### PR TITLE
return compressed commitment as root commitment

### PR DESCRIPTION
**Pull request by gballet**
This is the last missing piece before pbss/replay-fixes can be merged into the pbss PR, so that everything can be collapsed into a PR that is ready for rebase.

### CODE REVIEW

File: trie/verkle.go [hunk 1]
```diff
@@ -275,7 +275,9 @@ func (trie *VerkleTrie) Commit(_ bool) (common.Hash, *NodeSet, error) {
 	}
 	batch.Write()
 
-	return common.BytesToHash(nodes[0].CommitmentBytes[:32]), nil, nil
+	// Serialize root commitment in compressed form, and only the root commitment
+	rootH := root.Hash().BytesLE()
```
Comment (jsign) [inline trie/verkle.go:279]: Uhm, note that `root.hash()` returns a `Fr` so technically there's no compressed or uncompressed option.

**Review by jsign — APPROVED** [review]

File: trie/verkle.go [hunk 2]
```diff
@@ -275,7 +275,9 @@ func (trie *VerkleTrie) Commit(_ bool) (common.Hash, *NodeSet, error) {
 	}
 	batch.Write()
 
-	return common.BytesToHash(nodes[0].CommitmentBytes[:32]), nil, nil
+	// Serialize root commitment in compressed form, and only the root commitment
```
Comment (gballet) [inline trie/verkle.go:278]:
```suggestion
	// Serialize root commitment in compressed form, and only the root commitment
	// TODO figure out if Hash() is the correct function to call, or if PointToHash() should
	// be called instead.
```

File: trie/verkle.go [hunk 3]
```diff
@@ -275,7 +275,11 @@ func (trie *VerkleTrie) Commit(_ bool) (common.Hash, *NodeSet, error) {
...
+	// TODO figure out if Hash() is the correct function to call, or if PointToHash() should
+	// be called instead.
```
Comment (gballet) [inline trie/verkle.go:280]:
```suggestion
```

File: trie/verkle.go (hunk 2 above)
Comment (gballet) [inline trie/verkle.go:278]:
```suggestion
	// Serialize root commitment form
```

**Pull request merged**